From db1cd41d8cd8f76c640e822c1696581fd7e655a6 Mon Sep 17 00:00:00 2001
From: Marius Bergmann <marius@yeai.de>
Date: Fri, 27 Jan 2017 23:38:59 +0100
Subject: [PATCH 1/2] Fix wrong type in vteseq_n_lookup declaration

The declared unsigned int type differed from the code generated by
gperf, where the second argument is of size_t type. This lead to
ambigous function calls. I fixed this by changing the type of the second
argument to the proper type.
---
 src/vteseq.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/vteseq.cc b/src/vteseq.cc
index 2330939..e0ac14e 100644
--- a/src/vteseq.cc
+++ b/src/vteseq.cc
@@ -3409,7 +3409,7 @@ vte_sequence_handler_iterm2_1337(VteTerminalPrivate *that, GValueArray *params)
 #define VTE_SEQUENCE_HANDLER(name) name
 
 static const struct vteseq_n_struct *
-vteseq_n_lookup (register const char *str, register unsigned int len);
+vteseq_n_lookup (register const char *str, register size_t len);
 #include"vteseq-n.cc"
 
 #undef VTE_SEQUENCE_HANDLER

From bc2dc3ee37133629f4f5b0713929bcce10997690 Mon Sep 17 00:00:00 2001
From: Marius Bergmann <marius@yeai.de>
Date: Fri, 27 Jan 2017 23:47:15 +0100
Subject: [PATCH 2/2] Fix 'missing sentinel' errors during test compilation

'NULL' is a 32 bit integer on 64 bit systems, which prevents the
compiler from accepting it as a sentinel for the (64 bit) char pointer
varargs list. This can be fixed by an explicit cast.
---
 src/vteconv.cc  | 2 +-
 src/vtetypes.cc | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/vteconv.cc b/src/vteconv.cc
index b78d392..5cb63e7 100644
--- a/src/vteconv.cc
+++ b/src/vteconv.cc
@@ -771,7 +771,7 @@ int
 main (int argc,
       char *argv[])
 {
-        g_test_init (&argc, &argv, NULL);
+        g_test_init (&argc, &argv, (char *)NULL);
 
         g_test_add_func ("/vte/conv/utf8/strlen", test_utf8_strlen);
         g_test_add_func ("/vte/conv/utf8/validate", test_utf8_validate);
diff --git a/src/vtetypes.cc b/src/vtetypes.cc
index 1365a29..8f38c9d 100644
--- a/src/vtetypes.cc
+++ b/src/vtetypes.cc
@@ -407,7 +407,7 @@ test_util_smart_fd(void)
 int
 main(int argc, char *argv[])
 {
-        g_test_init (&argc, &argv, NULL);
+        g_test_init (&argc, &argv, (char *)NULL);
 
         g_test_add_func("/vte/c++/grid/coords", test_grid_coords);
         g_test_add_func("/vte/c++/grid/span", test_grid_span);
